lib/fastlane/plugin/wpmreleasetoolkit/versioning/formatters/derived_build_code_formatter.rb

```diff
@@ -3,23 +3,34 @@
 module Fastlane
   module Wpmreleasetoolkit
     module Versioning
+      MAX_TOTAL_DIGITS = 8
+
       # The `DerivedBuildCodeFormatter` class is a specialized build code formatter for derived build codes.
       # It takes in an AppVersion object and derives a build code from it.
       class DerivedBuildCodeFormatter
-        # Initialize the formatter with a configurable prefix.
+        # Initialize the formatter with configurable prefix and digit counts.
         #
         # @param [String] prefix The prefix to use for the build code. Must be a single digit (0-9), or empty string / nil.
+        # @param [Integer] major_digits Number of digits for major version. Defaults to 2.
+        # @param [Integer] minor_digits Number of digits for minor version. Defaults to 2.
+        # @param [Integer] patch_digits Number of digits for patch version. Defaults to 2.
+        # @param [Integer] build_digits Number of digits for build number. Defaults to 2.
         #
-        def initialize(prefix: nil)
+        def initialize(prefix: nil, major_digits: 2, minor_digits: 2, patch_digits: 2, build_digits: 2)
           prefix ||= ''
           validate_prefix!(prefix)
+
+          @digit_counts = [major_digits, minor_digits, patch_digits, build_digits]
+          @digit_counts.each { |d| validate_digit_count!(d) }
+          validate_total_digits!(@digit_counts)
+
           @prefix = prefix.to_s
```

Comment:
💄  Nitpick just for the form, to have everything prefix-related grouped together (and at the start since `prefix` is the first parameter so probably the first one that makes sense to assign):

```suggestion
          prefix ||= ''
          validate_prefix!(prefix)
          @prefix = prefix.to_s

          @digit_counts = [major_digits, minor_digits, patch_digits, build_digits]
          @digit_counts.each { |d| validate_digit_count!(d) }
          validate_total_digits!(@digit_counts)
```

Reply:
I like it! 👍